src/index/TextIndexLiteralFilter.h

```diff
@@ -0,0 +1,69 @@
+// Copyright 2025, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Felix Meisen (fesemeisen@outlook.de)
+
+#ifndef QLEVER_SRC_INDEX_TRIPLEINTEXTINDEX_H
+#define QLEVER_SRC_INDEX_TRIPLEINTEXTINDEX_H
+
+#include <string>
+
+#include "parser/RdfParser.h"
+
+class TextIndexLiteralFilter {
+ public:
+  enum struct FilterType { AcceptMatching, DeclineMatching };
+
+  TextIndexLiteralFilter()
+      : regex_{std::make_unique<RE2>("(?s).*")},
+        filterType_{FilterType::AcceptMatching} {};
```

Comment:
Document, what the default behavior is (add everything, add nothing, .... also goes for the IndexBuilderMain comments.